### ISSUE TITLE
Transcript enhancements

### ISSUE THREAD

**Issue by kbwagner** [openopps/OpenOppsTasks#183]
Goal for completion 5/29/19

Issue: Users are getting confused on adding a transcript in Open Opps. They don't see that the new tab has opened and don't know how to get back.

Things to consider to improve: 
1) Could we automatically select ‘transcript” on USAJOBS when they go to upload? There are so many other document types to upload and transcripts is at the bottom of the second column and they get confused
-  in USAJOBS document type other than transcript selected, doesn’t come over to Open Opps but no message appears.

Resolution: This issue would have to be done on the USAJOBS side.   Wait and see if this becomes a bigger issue

2)  Confusion over new window for USAJOBS- add text to transcript section with instructions
•	Pop up modal with instructions?  To tell them they will get a new tab, steps to follow, then close the other tab and click refresh

Resolution: 
- modify the instructions on the page AND add a modal that explains the process
![image](https://user-images.githubusercontent.com/18709918/58044502-e8151d80-7b0d-11e9-8989-628773e0a107.png)
- Consider updating the language on the new user flow pages as well.
- Tickets 3538 and 3539

3) If only one transcript in Open Opps can we default the radio button on the refresh?  

Resolution: Yes, if there is only one we can default.  If there are any more, we can't assume.
- new ticket 3537

4) in 5/2/19 testing Khadija asked to also provide text on the page to indicate requirements for offical transcripts.

Resolution: Julie is checking with Khadijah on language.

5) Do we want the refresh transcript button to always show in case they keep going back to the USAJOBS window?

Resolution: Asking team on slack
- Decision is no

**Comment by bmcorum** [comment]
@kbwagner to answer number 1, this would require custom code added to USAJOBS with a custom URL that would then need to be used by Open Opportunities. This is a hard ask and a bunch of code for so little gain. We would need to get with Dan and/or Brody.

**Comment by kbwagner** [comment]
@dalenard21 The decision on content was:
1) Change the instructions on the upload page
![image](https://user-images.githubusercontent.com/18709918/58044632-3de9c580-7b0e-11e9-905e-d3d09ec9121f.png)

2) Create instructions for a modal page - you're going to get a new window, will be directed to USAJOBS, select document type of transcript, close your window, in Open Opps click refresh

Do you have time to work on content for this?  The timeline would be by next week so we could get the ticket in and have it ready for the next sprint planning Thurs 5/30.  thanks!

**Comment by kbwagner** [comment]
@dalenard21 Alex also suggested we could look at the text on the new user pages where it tells students what to do.  So throwing that out as well.

**Comment by kbwagner** [comment]
Here is what Khadija provided as the language they currently use on transcripts: 
This is the language we currently use in our announcements.

Transcripts: Your current or most recent transcript must be included in your online application per the instructions below.  Both official AND unofficial transcripts are acceptable as part of the application process.  However, if you are selected for an internship, you will be required to provide official transcripts before starting your internship.  Once your official transcript is officially received, should it indicate that you are not an eligible student, your internship offer will be rescinded.

**Comment by abudz19** [comment]
This whole paragraph doesn't seem necessary for the application process. I propose that we note for the application process that both official and unofficial transcripts are acceptable, and that the rest of the information if crucial, be added as a drawer here: https://opm.invisionapp.com/d/main#/console/15360465/344480823/preview

@dalenard21 ?

**Comment by kbwagner** [comment]
Since it's an upload...the transcript will always be unofficial. I think the key piece of info is "If you are selectged for an internship, you will be required to provide official transcripts before starting your internship."  and probably the sentence after that.  I think the info just needs to display somewhere is that correct @Piglady64 since I think you were the one having that conversation with K?

**Comment by Piglady64** [comment]
@kbwagner @abudz19 the whole paragraph isn't necessary. Khadijah just wants the students to know that they will be required to provide an official transcript before they start their internship.

**Comment by dalenard21** [comment]
@kbwagner @abudz19 @Piglady64 

What about this:

1. Update the instructional text under "Transcripts" to:
Please select or upload your most recent academic transcript. We accept unofficial transcripts. If you’re selected for an internship, you’ll need to provide an official transcript before you start.

2. Instructions for the modal:

Heading: Upload a transcript
Copy: To upload a transcript 
1.	Click Continue—we’ll send you to your profile in USAJOBS. This will be a new tab in your browser.
2.	Select an existing document that’s in your profile, or add a new document—you’ll get another pop-up window.
3.	Confirm the document you want to upload.
4.	Select Transcript as the document type.
5.	Click Upload and close this window.
6.	Go back to the browser tab that say Open Opportunities.
7.	Go to the Transcript section and click Refresh—you should see the new transcript you uploaded.
8.	Select the transcript you want to include in your application.
9.	Continue with your application.


Definitely need some feedback on the modal.  Are those instructions accurate?  Am I missing a step?

**Comment by kbwagner** [comment]
@dalenard21 - Thanks!

For the first one on the instructions on the page...I might add "We acccept unofficial transcripts as part of your application." to make it clear that you can give unofficial now but in the end we need official.  So it would be, 'Please select or upload your most recent academic transcript. We accept unofficial transcripts as part of your application. If you’re selected for an internship, you’ll need to provide an official transcript before you start."

On the modal - yes the steps are correct. The only one I question is #2 "Select an existing document in your profile"...since we're only talking about transcripts if they had one in their profile it would already show in Open Opps.  So that may be an unnecessary sentence.

@Piglady64 @tlorrison any feedback?

**Comment by Piglady64** [comment]
How about : Please select or upload your most recent academic transcript - this can be an unofficial transcript.  If you’re selected for an internship, you’ll need to provide an official transcript before you start.
The short sentence: "We accept unofficial transcripts" just sounds a little wonky by itself. I understand it s a complete sentence but if we just add it to the previous sentence I think it would sound a little better.


For #2 I would say something like: "Upload a new transcript to your USAJOBS profile." (Katrina is right if the transcript was already in their USAJOBS profile for them to select it would have shown up in OO for them to add to their application. They need to upload a new transcript to their USAJOBS profile and then go back to OO, refresh transcripts, select the one they just uploaded and save and continue.

I don't think #3 is necessary. The user clicks upload, chooses the document, chooses a document type  and clicks complete upload. There is no confirm step.

#5 the button in USAJOBS says "complete upload" 

1. Click "upload transcript" (this is what the button says)we’ll send you to your profile in USAJOBS. This will be a new tab in your browser.
2. Click "upload document" 
3. Choose the transcript you want to upload.
4. Change the document name if needed
5. Select "transcript" as the document type.
6. Click "complete upload" and close this tab. (I would use the same words you used above so if you say "This will open in a new window" in #1 then use window, if you are going to stick with "This will be a new tab in your browser." then use tab.

I hope this makes sense.

**Comment by dalenard21** [comment]
@Piglady64 @kbwagner @abudz19 

Here is the content for:
1. Adding a modal to explain the process of uploading a new transcript. You will get this modal when you click "Upload transcript" on the application screen. 
2. Updated instructional text under the "Transcript" section of the application to explain they'll have to provide an official transcript if they get the internship.

@abudz19 I assume the modal will include a button that says "Continue" (or Continue to USAJOBS) to then take the user to USAJOBS in a new tab.  And, I think it'd be okay to use the default modal style. https://usajobs.github.io/design-system/modals/
[Department-of-state-student-application-transcript-upload-instructions_v4_20190529.docx](https://github.com/openopps/OpenOppsTasks/files/3233306/Department-of-state-student-application-transcript-upload-instructions_v4_20190529.docx)

**Comment by kbwagner** [comment]
Closing - this has been completed by dev
